sw/host/rom_ext_image_tools/signer/src/main.rs

```diff
@@ -9,91 +9,188 @@
+    hex_str
+        .as_bytes()
+        .iter()
+        .skip(2)
+        .rev()
+        .map(|digit| match digit {
+            b'0'..=b'9' => Ok(digit - b'0'),
+            b'a'..=b'f' => Ok(digit - b'a' + 10),
+            b'A'..=b'F' => Ok(digit - b'A' + 10),
+            _ => bail!("Invalid hex digit: {} ({}).", *digit as char, digit),
+        })
+        .enumerate()
+        .try_for_each(|(i, digit)| -> Result<()> {
+            match i % 2 {
+                0 => bytes.push(digit?),
+                _ => *bytes.last_mut().unwrap() |= digit? << 4,
+            };
+            Ok(())
+        })?;
+    Ok(bytes)
+}
 
-    image.update_signature_field(&signature.bytes());
+/// Updates the manifest of an image.
+// TODO: This function must use a public key.
+fn update_image_manifest(image: &mut Image, key: impl AsRef<Path>) -> Result<()> {
+    let key = fs::read(&key).with_context(|| {
+        format!("Failed to read the key from `{}`.", key.as_ref().display())
+    })?;
+    let key =
+        PrivateKey::parse_from_der(&key).context("Failed to parse the key.")?;
+
+    image.set_manifest_field(
+        &manifest::ROM_EXT_SIGNATURE_KEY_PUBLIC_EXPONENT,
+        key.public_exponent_be().iter().rev().cloned(),
+    )?;
+    image.set_manifest_field(
+        &manifest::ROM_EXT_SIGNATURE_KEY_MODULUS,
+        key.public_modulus_be().iter().rev().cloned(),
+    )?;
+    image.set_manifest_field(
+        &manifest::ROM_EXT_MANIFEST_IDENTIFIER,
+        parse_hex_str("0x4552544f")?,
+    )?;
+
+    image.set_manifest_field(
+        &manifest::ROM_EXT_IMAGE_VERSION,
+        parse_hex_str("0xa8ad8002")?,
+    )?;
+
+    // TODO: Do we need these fields?
+    let extensions = [
+        (
+            &manifest::ROM_EXT_EXTENSION0_OFFSET,
+            &manifest::ROM_EXT_EXTENSION0_CHECKSUM,
+        ),
+        (
+            &manifest::ROM_EXT_EXTENSION1_OFFSET,
+            &manifest::ROM_EXT_EXTENSION1_CHECKSUM,
+        ),
+        (
+            &manifest::ROM_EXT_EXTENSION2_OFFSET,
+            &manifest::ROM_EXT_EXTENSION2_CHECKSUM,
+        ),
+        (
+            &manifest::ROM_EXT_EXTENSION3_OFFSET,
+            &manifest::ROM_EXT_EXTENSION3_CHECKSUM,
+        ),
+    ];
+    for fields in &extensions {
+        image.set_manifest_field(fields.0, parse_hex_str("0xdeadbeef")?)?;
+        image.set_manifest_field(fields.1, parse_hex_str("0x8badbeef")?)?;
+    }
+    image.set_manifest_field(
+        &manifest::ROM_EXT_USAGE_CONSTRAINTS,
+        std::array::IntoIter::new(
+            [0xa6; manifest::ROM_EXT_USAGE_CONSTRAINTS.size_bytes],
+        ),
+    )?;
+    image.set_manifest_field(
+        &manifest::ROM_EXT_PERIPHERAL_LOCKDOWN_INFO,
+        std::array::IntoIter::new(
+            [0xa5; manifest::ROM_EXT_PERIPHERAL_LOCKDOWN_INFO.size_bytes],
+        ),
+    )?;
+    image.set_manifest_field(
+        &manifest::ROM_EXT_IMAGE_TIMESTAMP,
+        parse_hex_str("0x7766554433221100")?,
+    )?;
```

Comment:
Change to zero.

Ultimately, this value should be set by configuration or a command-line setting (example use case: in platforms, we repurposed the timestamp field to be the CL number at which the firmware was built.  In some version control systems, a CL number is a reasonable substitute for a timestamp and having the option to use such a value here would yield a reproducible binary artifact).

Reply:
Done. 
> a reproducible binary artifact

Yes, that's why this doesn't use `SystemTime::now()` anymore.